Fixes #77 - PoseidonTree root hash count is fixed

vlopes11 opened the pull request.

vlopes11 committed:
Fixes #77 - PoseidonTree root hash count is fixed

CPerezz commented on src/tree/mod.rs:
```diff
@@ -101,14 +101,11 @@ where
-        Ok(storage_scalar.to_owned().into())
+        let branch = self
+            .poseidon_branch(0)?
+            .unwrap_or(PoseidonBranch::mock(&[], self.branch_depth as usize));
```
I don't think this is the expected behaviour of the function. 

If `self.poseidon_branch` failed, there's no reason at all to return something that is completely mocked. Since it simply does not represent anything related to the real tree/branch.

vlopes11 replied:
That unwrap is not for failing, but for being non-existent. In this case, its a `None`, not an `Err`. If the tree don't contain any elements, it will be `None`

CPerezz reviewed (changes requested):
LGTM, just that Result nit that I would like to check if follows the expected behaviour.

vlopes11 committed:
Replace `PoseidonTree::get` for `poseidon_branch`

Kelvin `get` behaves differently than the poseidon tree should.

It will return a variable height tree, and a poseidon tree must always
return a fixed count of levels, defined by its depth.

This way, to avoid confusion, the user can only work with poseidon
branches unless he specifically calls the inner implementation of the
tree that will return an immutable branch

vlopes11 committed:
Removal of bumped version, unreleased changes

CPerezz reviewed (approved):
LGTM.

Good job!
